### PR TITLE
feat: request cancellation

### PR DESCRIPTION
**Pull request by DorianMaliszewski**
Possibility to cancel a Promise with an enhanced promise type

### CODE REVIEW

File: packages/use-dataloader/src/__tests__/useDataLoader.tsx [hunk 1]
```diff
@@ -15,27 +25,29 @@ const initialProps = {
     }),
 }
 // eslint-disable-next-line react/prop-types
-const wrapper = ({ children }: { children: React.ReactNode }) => (
+const wrapper = ({ children }: { children?: React.ReactNode }) => (
```
Comment (vincentaudebert) [inline packages/use-dataloader/src/__tests__/useDataLoader.tsx:28]: Le declarer en VoidFunctionComponent?

**Comment by vincentaudebert** [comment]
I get this 

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
          at TestComponent (/Volumes/Projets/scaleway-lib/node_modules/@testing-library/react-hooks/lib/helpers/createTestHarness.js:22:5)
          at DataLoaderProvider (/Volumes/Projets/scaleway-lib/packages/use-dataloader/src/DataLoaderProvider.tsx:33:31)
          at wrapper (/Volumes/Projets/scaleway-lib/packages/use-dataloader/src/__tests__/useDataLoader.tsx:28:20)
          at Suspense
          at ErrorBoundary (/Volumes/Projets/scaleway-lib/node_modules/react-error-boundary/dist/react-error-boundary.cjs.js:40:35)

      123 |     async cacheKey => {
      124 |       try {
    > 125 |         dispatch(Actions.createOnLoading())
          |         ^
      126 |         const promise = method()
      127 |         cancelMethodRef.current = promise.cancel
      128 |         const result = await promise.then(res => res)
```

in the tests. Is there no way to avoid this?

File: packages/use-dataloader/src/__tests__/useDataLoader.tsx [hunk 2]
```diff
@@ -582,4 +608,73 @@ describe('useDataLoader', () => {
...
+    const { result, unmount } = renderHook<
+      UseDataLoaderHookProps,
+      UseDataLoaderResult
+    >(
+      props =>
+        useDataLoader(
+          props.key,
+          () => {
+            const promise = new Promise(resolve => {
+              setTimeout(() => resolve(true), 500)
```
Comment (adriengibrat) [inline packages/use-dataloader/src/__tests__/useDataLoader.tsx:623]: I know it already was the value used for other tests, but 500ms seems too long IMHO.

**Review by adriengibrat — DISMISSED** [review]

**Review by adriengibrat — COMMENTED** [review]
Why not using the native fetch abort mechanism?
> the fetch() promise rejects with an "AbortError" DOMException.

https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal

File: packages/use-dataloader/src/__tests__/useDataLoader.tsx (hunk 2 above)
Comment (DorianMaliszewski) [reply packages/use-dataloader/src/__tests__/useDataLoader.tsx:623]: Changed by 100ms that's better yes

**Comment by DorianMaliszewski** [comment]
> I'm not fan of `enhanced promise type` :
> 
> Why not using the native fetch abort mechanism?
> 
> > the fetch() promise rejects with an "AbortError" DOMException.
> 
> https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal

We should handle both Axios and Fetch behaviour so we need to keep something flexible to ensure a compatibility between both. May in the fututre we will directly integrate a custom hook for each ones.

**Review by adriengibrat — APPROVED** [review]

**Review by vincentaudebert — APPROVED** [review]
💯

**Review by chambo-e — APPROVED** [review]

**Pull request merged**